scipy/optimize/_differentialevolution.py

```diff
@@ -379,35 +382,41 @@ def __init__(self, func, bounds, args=(),
             raise ValueError("The population initialization method must be one"
                              "of 'latinhypercube' or 'random'")
 
-        self.population_energies = np.ones(
-            popsize * parameter_count) * np.inf
+        self.population_energies = np.ones(self.population_size) * np.inf
 
         self.disp = disp
 
     def init_population_lhs(self):
         """
-        Initializes the population with Latin Hypercube Sampling
-        Latin Hypercube Sampling ensures that the sampling of parameter space
-        is maximised.
+        Initializes the population with Latin Hypercube Sampling.
+        Latin Hypercube Sampling ensures that each parameter is uniformly
+        sampled over its range.
         """
-        samples = np.size(self.population, 0)
-        N = np.size(self.population, 1)
         rng = self.random_number_generator
 
-        # Generate the intervals
-        segsize = 1.0 / samples
+        # Each parameter range needs to be sampled uniformly. The scaled
+        # parameter range ([0, 1)) needs to be split into
+        # `self.population_size` segments, each of which has the following
+        # size:
+        segsize = 1.0 / self.population_size
 
-        # Fill points uniformly in each interval
-        rdrange = rng.rand(samples, N) * segsize
-        rdrange += np.atleast_2d(
-            np.linspace(0., 1., samples, endpoint=False)).T
+        # Within each segment we sample from a uniform random distribution.
+        # We need to do this sampling for each parameter.
+        samples = segsize * rng.rand(self.population_size,
+                                     self.parameter_count)
 
-        # Make the random pairings
-        self.population = np.zeros_like(rdrange)
+        # Offset each segment to cover the entire parameter range [0, 1)
+        samples += np.atleast_2d(
+            np.linspace(0., 1., self.population_size, endpoint=False)).T
```

Comment:
I could undoubtedly be done in one line.  However, one of the comments made in #5101 was that the source code was difficult to follow (commenting, etc). 

This line is split into two to try and make it clearer what the `samples` array is and how it is created. I actually felt that the code as it existed was clear already, but then I originally wrote the code. 

I think that two lines is clearer, but I'm easy either way.